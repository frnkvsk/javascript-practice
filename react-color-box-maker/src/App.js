/**
 App - this component should render the BoxList component.
 */
import React from "react";
import BoxList from "./BoxList";

import "./App.css";


/** Simple app that just shows the LightsOut game. */

function App() {  
  // let [defaultProps, setProps] = useState("easy");
  // const setDefaultProps = dp => {
  //   defaultProps = dp;
  //   setProps(defaultProps);
  // }
  
  return (
    <div className="App">
      <BoxList />
      
    </div>
  );
}

export default App;
