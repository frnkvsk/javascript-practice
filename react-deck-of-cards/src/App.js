import React from 'react';
import DeckOfCards from './DeckOfCards';
import { makeStyles } from '@material-ui/core/styles';
import { Grid } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    background: '#c8e6c9',
    width: '100%',
    height: '100vh'
  },
}));

const App = () => {
  const classes = useStyles();
  
  return (
    <Grid container direction="column" className={classes.root} >
      <Grid item container>
        <Grid item xs={"auto"} sm={1} md={2} />
        <Grid item xs={12} sm={10} md={8} >
          <DeckOfCards />
        </Grid>      
        <Grid item xs={"auto"} sm={1} md={2} />
      </Grid>      
    </Grid>
  );
}

export default App;
