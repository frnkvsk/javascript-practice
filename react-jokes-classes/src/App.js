import React from "react";
import JokeListCLS from "./JokeListCLS";

function App() {
  return (
    <div className="App">
      <JokeListCLS />
    </div>
  );
}

export default App;
