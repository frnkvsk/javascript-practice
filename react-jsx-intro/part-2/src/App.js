import React from 'react';
import Tweet from './Tweet'

function App() {
  return (
    <div>
      <Tweet
        name="Harry Henderson"
        username="harryH"
        date={new Date().toDateString()}
        message="I am hairy"
      />
      <Tweet
        name="Mary Mary"
        username="marymary"
        date={new Date().toDateString()}
        message="very merry"
      />
      <Tweet
        name="Anderson Cooper"
        username="andyC"
        date={new Date().toDateString()}
        message="Stay tuned"
      />
    </div>
  );
}

export default App;
