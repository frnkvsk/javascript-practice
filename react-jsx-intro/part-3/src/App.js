import React from 'react';
import Person from './Person';

function App() {
  return (
    <>
    <Person 
      name="Alice"
      age={17}
      hobbies={["golf", "foozball", "painting"]}
    />    
    <Person 
      name="George"
      age={26}
      hobbies={["walking", "running", "jogging"]}
    />    
    <Person 
      name="Anderson"
      age={44}
      hobbies={["Cheeseburgers", "chess", "poker"]}
    />
    </>    
  );
}

export default App;
