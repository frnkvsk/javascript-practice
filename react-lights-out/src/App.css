.App {
  width: 500px;
  text-align: center;
  background: lightgoldenrodyellow;
  border: 1px solid rgb(1, 153, 1, .1);
}
