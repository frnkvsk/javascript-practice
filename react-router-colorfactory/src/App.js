import React, {useState} from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import ColorList from './ColorList';
import ColorDisplay from './ColorDisplay';
import ColorForm from './ColorForm';
import useLocalStorage from './hooks';

function App(props) {
  
  let [name, setName] = useState("black");
  let [value, setValue] = useState("#000000");
  let [state, setState] = useLocalStorage('colors', props);

  const clickHandlerList = newName => {    
    setName(name = newName);
  }
  const clickHandlerForm = (newName, newValue) => {       
    setName(name = newName);
    setValue(value = newValue);
    if(!state[name]) {
      setState(
        [...state,
        {'name': name, 'value': value}]
      )
    }
  }
  return (
    <div className="App">
      <BrowserRouter>
        <Switch>
          <Route exact path="/colors" >
            <ColorList state={state} clickHandler={clickHandlerList}/> 
          </Route>
          <Route exact path="/colors/new" >
            <ColorForm clickHandlerForm={clickHandlerForm}/> 
          </Route>
          <Route path="/colors/:color" >
            <ColorDisplay state={state.filter(e => e.name===name)[0]} /> 
          </Route>
          <Redirect to="/colors" />
        </Switch>
      </BrowserRouter>   
    </div>
  );
}
App.defaultProps = {
  colors: [{name: 'red', value: 'red'} , {name: 'green', value: 'green'} , {name: 'blue', value: 'blue'}]
}
export default App;
