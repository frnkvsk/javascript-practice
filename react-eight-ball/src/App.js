import React from 'react';
import EightBall from './EightBall';
import ColorBox from './ColorBox';
import './App.css';
function App() {
  return (
    <div className="App">
      <EightBall />
      <ColorBox />
    </div>
  );
}

export default App;
