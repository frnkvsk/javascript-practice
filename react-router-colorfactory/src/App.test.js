import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from './App';

// smoke and appearance tests
test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const colorForm = getByText(/welcome to the color factory/i);
  expect(colorForm).toBeInTheDocument();
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

// test complete display of color page
it("handles ColorDisplay click events", () => {
  // get link to display page and fire event to open display page
  var { getByText } = render(<App />);
  const red = getByText("red");
  expect(red).toBeInTheDocument();
  fireEvent.click(red);
  // verify display page is open
  var { getAllByText } = render(<App />);
  const redDisplay = getAllByText("ISN'T IT BEAUTIFUL?");
  expect(redDisplay[0]).toBeInTheDocument();
  const goBack = getAllByText("GO BACK");
  // click on go link to take us back to the home page
  fireEvent.click(goBack[0]);
  // verify we are now on home page
  var { getAllByText } = render(<App />);
  const green = getAllByText("green");
  expect(green[0]).toBeInTheDocument();
});

// test complete add a color form process
it("handles ColorForm click events", () => {
  // get link to form and fire event to open form
  var { getByText } = render(<App />);
  const addAColor = getByText(/add a color/i);
  expect(addAColor).toBeInTheDocument();
  fireEvent.click(addAColor);
  // verify form is open
  var { getAllByText } = render(<App />);
  const addThisColor = getAllByText(/add this color/i);
  expect(addThisColor[0]).toBeInTheDocument();
  const colorName = getAllByText(/color name/i);
  expect(colorName[0]).toBeInTheDocument();
  // add a new color text to the form input field and fire button click event
  fireEvent.change(colorName[0].nextSibling, {target : {value : 'purple'}});
  fireEvent.click(addThisColor[0]);
  // should take us back to the home page and new color should be present
  var { getAllByText } = render(<App />);
  const purple = getAllByText("purple");
  expect(purple[0]).toBeInTheDocument();
});