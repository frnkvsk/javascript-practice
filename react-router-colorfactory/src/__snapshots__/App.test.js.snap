// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="makeStyles-root-6"
    >
      <header
        class="makeStyles-header-7"
      >
        <h1>
          Welcome to the color factory.
        </h1>
        <a
          class="makeStyles-link1-9"
          href="/colors/new"
        >
          Add a color.
        </a>
      </header>
      <div
        class="makeStyles-main-8"
      >
        <h2>
          Please select a color
        </h2>
        <a
          class="makeStyles-link2-10"
          href="/colors/red"
        >
          red
        </a>
        <a
          class="makeStyles-link2-10"
          href="/colors/green"
        >
          green
        </a>
        <a
          class="makeStyles-link2-10"
          href="/colors/blue"
        >
          blue
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
